notebook/serverextensions.py

```diff
@@ -0,0 +1,197 @@
+    )
+    module, server_exts = _get_server_extension_metadata(package)
+    for server_ext in server_exts:
+        require = server_ext['require']
+        if require not in server_extensions:
+            server_extensions.append(require)
+            diff = {'NotebookApp': {'server_extensions': server_extensions}}
+    _recursive_update(data, diff)
+    _write_config_data(data, user=user, sys_prefix=sys_prefix)
+
+
+def disable_server_extension_python(package, user=False, sys_prefix=False):
+    """Disable a server extension associated with a Python package."""
+    data = _read_config_data(user=user, sys_prefix=sys_prefix)
+    server_extensions = (
+        data.setdefault("NotebookApp", {})
+        .setdefault("server_extensions", [])
+    )
+    module, server_exts = _get_server_extension_metadata(package)
+    for server_ext in server_exts:
+        require = server_ext['require']
+        if require in server_extensions:
+            server_extensions.remove(require)
+            diff = {'NotebookApp': {'server_extensions': server_extensions}}
+    _recursive_update(data, diff)
+    _write_config_data(data, user=user, sys_prefix=sys_prefix)
+
+# ----------------------------------------------------------------------
+# Applications
+# ----------------------------------------------------------------------
+
+
+class ToggleServerExtensionApp(ToggleNBExtensionApp):
+
+    name = "jupyter serverextension enable/disable"
+    description = "Enable/disable a server extension using frontend configuration files."
+
+    def _toggle_server_extension(self, require):
+        config_dir = _get_config_dir(user=self.user, sys_prefix=self.sys_prefix)
+        cm = BaseJSONConfigManager(parent=self, config_dir=config_dir)
+        cfg = cm.get("jupyter_notebook_config")
+        server_extensions = (
+            cfg.setdefault("NotebookApp", {})
+            .setdefault("server_extensions", [])
+        )
+        if self._toggle_value:
+            if require not in server_extensions:
+                server_extensions.append(require)
+            else:
+                print("server extension already enabled")
+        elif self._toggle_value is None:
+            if require not in server_extensions:
+                print("server extension already disabled")
+            else:
+                server_extensions.remove(require)
+        cm.update("jupyter_notebook_config", cfg)
+
+    def toggle_server_extension_python(self, package):
+        m, server_exts = _get_server_extension_metadata(package)
+        for server_ext in server_exts:
+            require = server_ext['require']
+            self._toggle_server_extension(require)
+
+    def toggle_server_extension(self, require):
+        self._toggle_server_extension(require)
+
+    def start(self):
+
+        if not self.extra_args:
+            self.log.warn('Please specify a server extension/package to enable or disable')
+            sys.exit(1)
+        elif len(self.extra_args) > 1:
```

Comment:
Not sure, I think so.  I also see another `self.log.warn` + `sys.exit` , I can fix that too.  To be honest, this is my first time looking at serverextensions.py too :P